### PR TITLE
Starting update in routing.rst

### PR DESCRIPTION
**Pull request by tavaresgerson**

### CODE REVIEW

File: pt/development/routing.rst [hunk 1]
```diff
@@ -1,40 +1,20 @@
...
-Routing in CakePHP also encompasses the idea of reverse routing, where an array
-of parameters can be transformed into a URL string. By using reverse routing,
-you can re-factor your application's URL structure without having to update all
-your code.
+O roteamento no CakePHP também abrange a idéia de roteamento reverso, onde uma matriz de parâmetros pode ser transformada em uma string de URL. Ao usar o roteamento reverso, você pode redimensionar a estrutura de URL do seu aplicativo sem precisar atualizar todo o seu código.
 
 .. index:: routes.php
 
-Quick Tour
+Tour rápido
```
Comment (markstory) [inline pt/development/routing.rst]: Title underlines need to be the same length as the text above them.

**Comment by markstory** [comment]
@tavaresgerson Let me know when you are ready to have this merged.

**Comment by tavaresgerson** [comment]
Done! 😅

File: pt/development/routing.rst [hunk 2]
```diff
@@ -499,83 +380,78 @@ like below and CakePHP will know how to form the URL as defined in the routes::
...
-    ``_name`` cannot be used twice, even if the names occur inside a different
-    routing scope.
+    Os nomes das rotas devem ser exclusivos em todo o aplicativo. O mesmo 
+    ``_name`` não pode ser usado duas vezes, mesmo que os nomes ocorram dentro de um 
+    escopo de roteamento diferente.
 
-When building named routes, you will probably want to stick to some conventions
-for the route names. CakePHP makes building up route names easier by allowing
-you to define name prefixes in each scope::
+Ao criar rotas nomeadas, você provavelmente desejará seguir algumas convenções para os nomes das rotas. O CakePHP facilita a criação de nomes de rotas, permitindo definir prefixos de nomes em cada escopo:
```
Comment (markstory) [inline pt/development/routing.rst]:
```suggestion
Ao criar rotas nomeadas, você provavelmente desejará seguir algumas convenções para os nomes das rotas. O CakePHP facilita a criação de nomes de rotas, permitindo definir prefixos de nomes em cada escopo::
```

Need another `:` to make the following block a code section.

File: pt/development/routing.rst [hunk 3]
```diff
@@ -499,83 +380,78 @@ like below and CakePHP will know how to form the URL as defined in the routes::
...
-    // Or with prefix()
+    // Ou com prefix()
     Router::prefix('Admin', ['_namePrefix' => 'admin:'], function ($routes) {
-        // Connect routes.
+        // Conecta rotas.
     });
 
-You can also use the ``_namePrefix`` option inside nested scopes and it works as
-you'd expect::
+Você também pode usar a opção ``_namePrefix`` dentro de escopos aninhados e funciona conforme o esperado:
```
Comment (markstory) [inline pt/development/routing.rst]:
```suggestion
Você também pode usar a opção ``_namePrefix`` dentro de escopos aninhados e funciona conforme o esperado::
```

File: pt/development/routing.rst [hunk 4]
```diff
@@ -585,183 +461,155 @@ Prefix Routing
 
 .. php:staticmethod:: prefix($name, $callback)
 
-Many applications require an administration section where
-privileged users can make changes. This is often done through a
-special URL such as ``/admin/users/edit/5``. In CakePHP, prefix routing
-can be enabled by using the ``prefix`` scope method::
+Muitos aplicativos requerem uma seção de administração na qual usuários privilegiados podem fazer alterações. Isso geralmente é feito por meio de uma URL especial, como ``/admin/users/edit/5``. No CakePHP, o roteamento de prefixo pode ser ativado usando o método de escopo ``prefix``:
```
Comment (markstory) [inline pt/development/routing.rst]:
```suggestion
Muitos aplicativos requerem uma seção de administração na qual usuários privilegiados podem fazer alterações. Isso geralmente é feito por meio de uma URL especial, como ``/admin/users/edit/5``. No CakePHP, o roteamento de prefixo pode ser ativado usando o método de escopo ``prefix``::
```

File: pt/development/routing.rst [hunk 5]
```diff
@@ -585,183 +461,155 @@ Prefix Routing
...
             $routes->connect('/:controller');
         });
     });
 
-The above would create a route template like ``/debug_kit/admin/:controller``.
-The connected route would have the ``plugin`` and ``prefix`` route elements set.
+O exemplo acima criaria um modelo de rota como ``/debug_kit/admin/:controller``. A rota conectada teria os elementos de rota ``plugin`` e ``prefix`` definidos.
 
-When defining prefixes, you can nest multiple prefixes if necessary::
+Ao definir prefixos, você pode aninhar vários prefixos, se necessário:
```
Comment (markstory) [inline pt/development/routing.rst]:
```suggestion
Ao definir prefixos, você pode aninhar vários prefixos, se necessário::
```

File: pt/development/routing.rst [hunk 6]
```diff
@@ -585,183 +461,155 @@ Prefix Routing
...
-        // Routes connected here are prefixed with '/debug_kit' and
-        // have the plugin route element set to 'DebugKit'.
+        // As rotas conectadas aqui são prefixadas com '/debug_kit' e 
+        // têm o elemento de rota do plug-in definido como 'DebugKit'.
         $routes->connect('/:controller');
     });
 
-When creating plugin scopes, you can customize the path element used with the
-``path`` option::
+Ao criar escopos de plug-in, você pode personalizar o caminho usando a opção ``path``:
```
Comment (markstory) [inline pt/development/routing.rst]:
```suggestion
Ao criar escopos de plug-in, você pode personalizar o caminho usando a opção ``path``::
```

File: pt/development/routing.rst [hunk 7]
```diff
@@ -585,183 +461,155 @@ Prefix Routing
...
+Você pode criar links que apontam para um plug-in, adicionando a chave do plug-in a seu array de URL::
 
     echo $this->Html->link(
         'New todo',
         ['plugin' => 'Todo', 'controller' => 'TodoItems', 'action' => 'create']
     );
 
-Conversely if the active request is a plugin request and you want to create
-a link that has no plugin you can do the following::
+Por outro lado, se a solicitação ativa for uma solicitação de plug-in e você desejar criar um link que não possua plug-in, faça o seguinte:
```
Comment (markstory) [inline pt/development/routing.rst]:
```suggestion
Por outro lado, se a solicitação ativa for uma solicitação de plug-in e você desejar criar um link que não possua plug-in, faça o seguinte::
```

File: pt/development/routing.rst [hunk 8]
```diff
@@ -585,183 +461,155 @@ Prefix Routing
...
         $routes->fallbacks(DashedRoute::class);
     });
 
-Matching Specific HTTP Methods
-------------------------------
+Correspondendo a métodos HTTP específicos
+-----------------------------------------
 
-Routes can match specific HTTP methods using the HTTP verb helper methods::
+As rotas podem corresponder a métodos HTTP específicos usando os métodos auxiliares de verbo HTTP:
```
Comment (markstory) [inline pt/development/routing.rst]:
```suggestion
As rotas podem corresponder a métodos HTTP específicos usando os métodos auxiliares de verbo HTTP::
```

File: pt/development/routing.rst [hunk 9]
```diff
@@ -771,136 +619,123 @@ Routes can match specific HTTP methods using the HTTP verb helper methods::
         )->setMethods(['POST', 'PUT']);
     });
 
-You can match multiple HTTP methods by using an array. Because the ``_method``
-parameter is a routing key, it participates in both URL parsing and URL
-generation. To generate URLs for method specific routes you'll need to include
-the ``_method`` key when generating the URL::
+Você pode combinar vários métodos HTTP usando uma matriz. Como o parâmetro ``_method`` é uma chave de roteamento, ele participa da análise e geração de URLs. Para gerar URLs para rotas específicas de métodos, você precisará incluir a chave ``_method`` ao gerar a URL:
```
Comment (markstory) [inline pt/development/routing.rst]:
```suggestion
Você pode combinar vários métodos HTTP usando uma matriz. Como o parâmetro ``_method`` é uma chave de roteamento, ele participa da análise e geração de URLs. Para gerar URLs para rotas específicas de métodos, você precisará incluir a chave ``_method`` ao gerar a URL::
```

File: pt/development/routing.rst [hunk 10]
```diff
@@ -771,136 +619,123 @@ Routes can match specific HTTP methods using the HTTP verb helper methods::
...
             '/images/old-log.png',
             ['controller' => 'Images', 'action' => 'oldLogo']
         )->setHost('images.example.com');
     });
 
-The ``_host`` option is also used in URL generation. If your ``_host`` option
-specifies an exact domain, that domain will be included in the generated URL.
-However, if you use a wildcard, then you will need to provide the ``_host``
-parameter when generating URLs::
+A opção ``_host`` também é usada na geração de URL. Se a opção ``_host`` especificar um domínio exato, esse domínio será incluído no URL gerado. No entanto, se você usar um curinga, precisará fornecer o parâmetro ``_host`` ao gerar URLs:
```
Comment (markstory) [inline pt/development/routing.rst]:
```suggestion
A opção ``_host`` também é usada na geração de URL. Se a opção ``_host`` especificar um domínio exato, esse domínio será incluído no URL gerado. No entanto, se você usar um curinga, precisará fornecer o parâmetro ``_host`` ao gerar URLs::
```

File: pt/development/routing.rst [hunk 11]
```diff
@@ -1127,63 +955,58 @@ You can provide ``connectOptions`` key in the ``$options`` array for
...
+--------------------------------------
 
-By default, multi-worded controllers' URL fragments are the underscored
-form of the controller's name. E.g., ``BlogPostsController``'s URL fragment
-would be **/blog_posts**.
+Por padrão, os fragmentos de URL dos controladores com várias palavras são a forma sublinhada do nome do 
+controlador. Por exemplo, fragmento de URL do ``BlogPostsController`` seria **/blog_posts**.
 
-You can specify an alternative inflection type using the ``inflect`` option::
+Você pode especificar um tipo de inflexão alternativo usando a opção ``inflect``:
```
Comment (markstory) [inline pt/development/routing.rst]:
```suggestion
Você pode especificar um tipo de inflexão alternativo usando a opção ``inflect``::
```

**Comment by tavaresgerson** [comment]
Sorry, I'll have attention next time 😸

**Comment by markstory** [comment]
Thank you 👍

**Pull request merged**